File: ios/UnmatchedCounter/Model/PlayerSetupModel.swift
import Foundation

struct PlayerSetupModel: Equatable {
    let id: Int
    var characterName: String
    var color: PlayerColor
    var startingLifeSegments: [Int]

    init(id: Int = 0,
         characterName: String = UnmatchedCharacters.names.first ?? "",
         color: PlayerColor = .none,
         startingLifeSegments: [Int] = [GameRules.startingLife]) {
        precondition(!startingLifeSegments.isEmpty &&
                     startingLifeSegments.count <= LifeSegmentLimits.maxLifeSegments,
                     "startingLifeSegments must have 1..\(LifeSegmentLimits.maxLifeSegments) values")
        self.id = id
        self.characterName = characterName
        self.color = color
        self.startingLifeSegments = startingLifeSegments
    }
}

File: app/src/main/java/com/unmatchedcounter/model/player/PlayerSetupModel.kt
package com.unmatchedcounter.model.player

import android.os.Parcelable
import com.unmatchedcounter.GameRules
import com.unmatchedcounter.UnmatchedCharacters
import com.unmatchedcounter.util.LifeSegmentUtils
import kotlinx.parcelize.Parcelize

@Parcelize
data class PlayerSetupModel(
    val id: Int = 0,
    val characterName: String = UnmatchedCharacters.NAMES.first(),
    val color: PlayerColor = PlayerColor.NONE,
    val startingLifeSegments: List<Int> = listOf(GameRules.STARTING_LIFE),
) : Parcelable {

    init {
        require(startingLifeSegments.isNotEmpty() && startingLifeSegments.size <= LifeSegmentUtils.MAX_LIFE_SEGMENTS) {
            "startingLifeSegments must have 1..${LifeSegmentUtils.MAX_LIFE_SEGMENTS} values"
        }
    }
}
